addons/account/models/chart_template.py

```diff
@@ -297,17 +295,21 @@ def tax_template_changed(tax, template):
             for xmlid, values in records.items():
                 if model_name == 'account.fiscal.position':
                     # Only add tax mappings containing new taxes
-                    values['tax_ids'] = [
-                        (command, id, vals)
-                        for command, id, vals in values.get('tax_ids', [])
-                        if (
-                            command not in (Command.UPDATE, Command.CREATE)
-                            or not self.ref(vals['tax_src_id'], raise_if_not_found=False)
-                            or (vals.get('tax_dest_id') and not self.ref(vals['tax_dest_id'], raise_if_not_found=False))
-                        )
-                    ]
-                    if not values['tax_ids']:
-                        del values['tax_ids']
+                    if old_tax_ids := values.pop('tax_ids', []):
+                        new_tax_ids = []
+                        for element in old_tax_ids:
+                            match element:
+                                case Command() as command, _, dict() as subvals:
+                                    src_id, dest_id = subvals.get('tax_src_id'), subvals.get('tax_dest_id')
+                                    if (
+                                        command.value not in (Command.CREATE, Command.UPDATE)
+                                        or not self.ref(src_id, raise_if_not_found=False)
+                                        or (dest_id and not self.ref(dest_id, raise_if_not_found=False))
+                                    ):
+                                        new_tax_ids.append(element)
```

Comment:
Fine with the `CREATE` only.

Yet, putting the `tax_src_id` and `tax_dest_id` there, makes them mandatory for the match, so it's not logically equivalent...(?) In case we want this, then the `or (dest_id and` part would no longer be useful. I don't know why someone would come up with a half-filled rule for fiscal positions. Is that a use-case?

Reply:
The `dest_id` would then be put at `False` explicitly, for instance in the CSV for Spain:

| id | name | tax_ids/tax_src_id | tax_ids/tax_dest_id |
| - | - | - | - |
| fp_extra | Extra-community | account_tax_template_p_iva0_s_bc | account_tax_template_p_iva0_ibc |
|  |  | account_tax_template_p_iva0_s_sc | account_tax_template_p_iva0_isc |
|  |  | account_tax_template_p_iva4_bc |  |